tests/run.rs

```diff
@@ -800,3 +800,122 @@ fn complains_when_expected_an_unmocked_command() -> R<()> {
+                    |      - /bin/cat /foo/bar
+                    |    filetree:
+                    |      - /foo/bar
+                "##,
+                Ok(()),
+            )?;
+            Ok(())
+        }
+
+        #[test]
+        fn allows_to_mock_folders_existence_in_ruby() -> R<()> {
+            test_run(
+                r##"
+                    |#!/usr/bin/env ruby
+                    |`cat /foo/bar/test` if Dir.exists?("/foo/bar/")
+                    |`cat /foo/bar/test2` if File.directory?("/foo/bar/")
+                "##,
+                r##"
+                    |protocols:
+                    |  - protocol:
+                    |      - /bin/cat /foo/bar/test
+                    |      - /bin/cat /foo/bar/test2
+                    |    filetree:
+                    |      - /foo/bar/
+                "##,
+                Ok(()),
+            )?;
+            Ok(())
+        }
+    }
+
+    mod in_bash {
+        use super::*;
+
+        #[test]
+        fn allows_to_mock_files_existence() -> R<()> {
+            test_run(
+                r##"
+                    |#!/usr/bin/env bash
+                    |if [ -f /foo/bar ]; then
+                    |  cat /foo/bar
+                    |fi
+                "##,
+                r##"
+                    |protocols:
+                    |  - protocol:
+                    |      - /bin/cat /foo/bar
+                    |    filetree:
+                    |      - /foo/bar
+                "##,
+                Ok(()),
+            )?;
+            Ok(())
+        }
+
+        #[test]
+        fn allows_to_mock_directory_existence() -> R<()> {
+            test_run(
+                r##"
+                    |#!/usr/bin/env bash
+                    |if [ -d /foo/bar/ ]; then
+                    |  cat /foo/bar/test
+                    |fi
+                "##,
+                r##"
+                    |protocols:
+                    |  - protocol:
+                    |      - command: /bin/cat /foo/bar/test
+                    |    filetree:
+                    |      - /foo/bar/
+                "##,
+                Ok(()),
+            )?;
+            Ok(())
+        }
+
+        #[test]
+        fn does_not_mock_existence_of_unspecified_files() -> R<()> {
+            test_run(
+                r##"
+                    |#!/usr/bin/env bash
+                    |if [ -f /foo/bar ]; then
+                    |  cat /foo/bar
+                    |fi
+                "##,
+                r##"
+                    |protocols:
+                    |  - protocol:
+                    |    - /bin/cat /foo/bar
+                "##,
+                Err(&trim_margin(
+                    "
+                        |error:
+                        |  expected: /bin/cat /foo/bar
+                        |  received: <script terminated>
+                    ",
+                )?),
+            )?;
+            Ok(())
+        }
+    }
+}
```

Comment:
@matthandlersux: I would change these integration tests a bit, but I think it'd be easier to talk about it in person, if you're up for that.

Reply:
@matthandlersux: Since we didn't get a chance to talk I left some comments. I'll be up for looking at the changed integration tests when you've addressed them.